packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -933,9 +934,50 @@ export class GarbageCollector implements IGarbageCollector {
 	 * cases where objects are used after they are deleted and throw / log errors accordingly.
 	 */
 	public isNodeDeleted(nodePath: string): boolean {
+		//* TODO: Replace usages with getInternalReferenceInfo?
 		return this.deletedNodes.has(nodePath);
 	}
 
+	/**
+	 * Get this client's estimation of the reference state of the give node
+	 * It may be inaccurate in some cases, e.g. if an interactive client sees an object as unreferenced on load,
+	 * some time later it will transition to Inactive even if it is revived in the meantime.
+	 * @param nodePath - Container-relative path to the node
+	 */
+	public getInternalReferenceInfo(nodePath: string): IFluidInternalReferenceInfo {
+		//* Is this a fair requirement?
```

Comment:
I believe so. Ideally, if a data store is deleted, it's context shouldn't exist and we won't be here.